### ISSUE TITLE
allow val_check_interval to be a float

### ISSUE THREAD

**Issue by wsnoble** [Noble-Lab/casanovo#627]
In pytorch, val_check_interval can be either a float in the range [0,1] or an int.  If it's a float, it specifies how frequently the validation set is analyzed per epoch, whereas if it's an int it specifies the same thing in number of batches (steps).  Currently, casanovo only allows it to be an int. I think we should allow either, and document it as such in the config file.  Furthermore, I think the default value should be 1.0.

**Comment by ssiddhantsharma** [comment]
I've opened #682, which lets `val_check_interval` accept a float in `[0, 1]` (fraction of an epoch) or an int (steps), with the default set to `1.0`. It also sets `check_val_every_n_epoch=1` when the value is a float, which Lightning requires.
